lang/src/org/partiql/lang/eval/ExprValueExtensions.kt

```diff
@@ -709,6 +709,8 @@ fun Sequence<ExprValue>.distinct(): Sequence<ExprValue> {
                 yield(it)
             }
         }
+        // clear out the seen set in case we need to call distinct multiple times, i.e. Type Inferencer
+        seen.clear()
```

Comment:
An extension function becomes a static function in Java, so the TreeSet was being instantiated in the static class which caused an unintentional global. It should be moved inside the sequence creation

```suggestion
fun Sequence<ExprValue>.distinct() = sequence {
    val seen = TreeSet(DEFAULT_COMPARATOR)
    this@distinct.forEach {
        if (!seen.contains(it)) {
            seen.add(it.unnamedValue())
            yield(it)
        }
    }
}
```

Reply:
The nearby multiplicities function will also need to be fixed

Reply:
The multiplicities function should work fine based on the examination of the compiled java code. 

```
   public static final TreeMap multiplicities(@NotNull Sequence $this$multiplicities) {
      Intrinsics.checkNotNullParameter($this$multiplicities, "$this$multiplicities");
      // Seems like the a new treeMap is constructed every time the method got called. 
      TreeMap multiplicities = new TreeMap((Comparator)DEFAULT_COMPARATOR);
      int $i$f$forEach = false;
      Iterator var4 = $this$multiplicities.iterator();

      while(var4.hasNext()) {
         Object element$iv = var4.next();
         ExprValue it = (ExprValue)element$iv;
         int var7 = false;
         multiplicities.compute(it, (BiFunction)ExprValueExtensionsKt$multiplicities$1$1.INSTANCE);
      }

      return multiplicities;
   }
```